test/suites/dev/moonbase/test-precompile/test-precompile-pallet-xcm-2.ts

```diff
@@ -0,0 +1,472 @@
+import "@moonbeam-network/api-augment";
+import {
+  beforeAll,
+  describeSuite,
+  fetchCompiledContract,
+  expect,
+  customDevRpcRequest,
+} from "@moonwall/cli";
+import {
+  ALITH_ADDRESS,
+  BALTATHAR_ADDRESS,
+  alith,
+  baltathar,
+  createEthersTransaction,
+  PRECOMPILE_NATIVE_ERC20_ADDRESS,
+} from "@moonwall/util";
+import type { u128 } from "@polkadot/types-codec";
+import type { PalletAssetsAssetAccount, PalletAssetsAssetDetails } from "@polkadot/types/lookup";
+import { checksumAddress, encodeFunctionData } from "viem";
+import {
+  expectEVMResult,
+  mockOldAssetBalance,
+  PARA_1000_SOURCE_LOCATION,
+  registerOldForeignAsset,
+  mockHrmpChannelExistanceTx,
+} from "../../../../helpers";
+
+import type { AssetMetadata } from "../../../../helpers";
+
+const PRECOMPILE_PALLET_XCM_ADDRESS: `0x${string}` = "0x000000000000000000000000000000000000081A";
+
+export const para1000AssetMetadata: AssetMetadata = {
+  name: "PARA",
+  symbol: "PARA",
+  decimals: 12n,
+  isFrozen: false,
+};
+
+describeSuite({
+  id: "D012901",
+  title: "Precompiles - PalletXcm: Native fee",
+  foundationMethods: "dev",
+  testCases: ({ context, it }) => {
+    let relayAssetId: u128;
+    let para1000AssetId: u128;
+    let ADDRESS_PARA_1000_ERC20: string;
+    const destinationPara = 1000;
+    const ADDRESS_RELAY_ERC20 = "0xFfFFfFff1FcaCBd218EDc0EbA20Fc2308C778080";
+    const RELAY_ASSET_ID = 42259045809535163221576417993425387648n;
+    const amountToSend = 100n;
+
+    beforeAll(async () => {
+      const balance = 200000000000000n;
+      const assetBalance: PalletAssetsAssetAccount = context
+        .polkadotJs()
+        .createType("PalletAssetsAssetAccount", {
+          balance: balance,
+        });
+      relayAssetId = context.polkadotJs().createType("u128", RELAY_ASSET_ID);
+
+      const assetDetails: PalletAssetsAssetDetails = context
+        .polkadotJs()
+        .createType("PalletAssetsAssetDetails", {
+          supply: balance,
+        });
+
+      await mockOldAssetBalance(
+        context,
+        assetBalance,
+        assetDetails,
+        alith,
+        relayAssetId,
+        ALITH_ADDRESS,
+        true
+      );
+
+      const { registeredAssetId } = await registerOldForeignAsset(
```

Comment:
We should start using the new foreign assets approach, since I will probably get rid of assetsManager pallet in the future

Reply:
Yeah I didn't take this approach initially as it is a bit out of the scope of this PR (which just adds more TS tests). 

Given that it could imply a bigger refactor to the pallet-xcm precompile, do you think it's worth doing it in a separate PR? Otherwise I can do it in this one.

Reply:
I think we should do it here, otherwise it will add technical debt to when we remove pallet asset-manager.

Reply:
Sure, will do it here then.